core/src/main/scala/cats/MonadReader.scala

```diff
@@ -0,0 +1,14 @@
+package cats
+
+/** A monad that has the ability to read from an environment. */
+trait MonadReader[F[_, _], R] extends Monad[F[R, ?]] {
+  /** Get the environment */
+  def ask: F[R, R]
+
+  /** Modify the environment */
+  def local[A](f: R => R)(fa: F[R, A]): F[R, A]
```

Comment:
Should we make this `def local[A](fa: F[R, A])(f: R => R): F[R, A]` to line up with the definitions for `map`, `flatMap`, `traverse`, etc in other type classes?